server/bleep/src/cache.rs

```diff
@@ -61,11 +61,22 @@ pub(crate) type FileCacheSnapshot = Arc<scc::HashMap<String, FreshValue<()>>>;
 pub(crate) struct FileCache<'a> {
     db: &'a SqlDb,
     reporef: &'a RepoRef,
+    semantic: Option<&'a Semantic>,
```

Comment:
I didn't really want to go into this whole story here, as it would be a fairly invasive change. I'd rather handle this in a separate PR that's dedicated to this.